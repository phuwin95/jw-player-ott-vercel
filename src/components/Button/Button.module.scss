@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';

$small-button-height: 28px;
$medium-button-height: 46px;
$large-button-height: 46px;

.button {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  height: $medium-button-height;
  padding: 0 16px;
  overflow: hidden;

  font-family: var(--body-alt-font-family);
  font-weight: var(--body-font-weight-bold);
  font-size: 17px;
  font-style: normal;
  line-height: 1.2;
  text-align: center;
  text-decoration: none;
  border-radius: 4px;
  outline: none;
  cursor: pointer;
  transition: background-color 0.1s ease, transform 0.1s ease;

  @media (hover: hover) and (pointer: fine) {
    &:not(.disabled) {
      &:hover,
      &:focus {
        z-index: 1;
        transform: scale(1.1);
      }

      &:focus:not(:focus-visible):not(:hover) {
        transform: scale(1);
      }

      &:focus-visible {
        transform: scale(1.1);
      }
    }
  }

  &.disabled {
    cursor: default;
    opacity: 0.5;
  }

  &.small {
    height: $small-button-height;
  }

  &.large {
    height: $large-button-height;
  }

  &.default {
    color: currentColor;
    background-color: theme.$btn-default-bg;
  }

  &.primary {
    color: var(--highlight-contrast-color, theme.$btn-primary-color);
    background: theme.$btn-primary-bg-gradient;
  }

  &.outlined {
    border: 1px solid rgba(255, 255, 255, 0.3);

    &:not(.disabled) {
      &.active,
      &:focus {
        color: var(--highlight-contrast-color, theme.$btn-primary-color);
        background-color: var(--highlight-color, theme.$btn-primary-bg);
        border-color: var(--highlight-color, theme.$btn-primary-bg);
      }
    }
  }

  &.text {
    background: none;
    opacity: 0.7;

    &:not(.disabled) {
      &.active,
      &:focus {
        opacity: 1;
      }

      &:hover {
        z-index: 1;
        background: theme.$btn-default-bg;
        opacity: 1;
      }
    }
  }

  &.danger {
    color: theme.$btn-danger-color;
    background-color: theme.$btn-danger-bg;
    outline: theme.$btn-danger-outline;
  }

  &.delete {
    color: theme.$btn-delete-color;
    background-color: theme.$btn-delete-bg;
    outline: none;
  }

  &.fullWidth {
    justify-content: center;
    width: 100%;

    @media (hover: hover) and (pointer: fine) {
      &:hover:not(.disabled) {
        transform: scale(1.04);
      }
    }
  }

  > svg {
    width: 18px;
    height: 18px;
  }
}

.startIcon {
  display: flex;
  align-items: center;
  height: 100%;
  margin-right: 6px;

  > svg {
    width: 20px;
    height: 20px;
    fill: currentColor;

    @include responsive.tablet-only() {
      width: 18px;
      height: 18px;
    }
  }
}

.hidden {
  visibility: hidden;
}

.centerAbsolute {
  position: absolute;
  right: 0;
  left: 0;
  margin: auto;
  transform: translate(-5px, -5px);
}
