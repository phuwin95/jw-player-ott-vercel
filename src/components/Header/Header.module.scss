@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';
@use 'src/styles/mixins/utils';

//
// Header container
//
.container {
  z-index: 3;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 51px;
  padding: 10px 0 8px;
  color: variables.$seiska-yellow;
  text-align: center;
  background-color: var(--body-background-color);
}

//
// Header menu
//
.menu {
  display: none;
}

//
// Header brand/logo
//
.iconButton {
  width: 36px;
  height: 36px;

  > svg {
    fill: currentColor;
  }
}

.mobileOnly {
  display: none;
  @include responsive.mobile-and-tablet() {
    display: inherit;
  }
}

//
// Header brand
//
.brand {
  align-self: center;
  margin-right: variables.$base-spacing;
}

//
// Header navigation
//
.nav {
  > a {
    height: 22px;
    min-height: 22px;
    margin-right: 22px;
    padding: 0 calc(#{variables.$base-spacing} / 2);
    font-weight: var(--body-font-weight-bold);
    font-size: 16px;
    line-height: 1.2;
    text-align: center;
  }
}

//
// Header actions
//
.actions {
  position: relative;
  display: flex;
  justify-content: flex-end;
}

//
// Container for the user button(s)
//
.buttonContainer {
  // this is a visual fix for putting a button with background besides a transparent button
  margin-left: variables.$base-spacing;

  > button:first-child {
    margin-right: calc(#{variables.$base-spacing} / 2);
  }
}

.userMenuButton {
  margin-left: 8px;
}

.brandCentered {
  .brand {
    text-align: center;
  }
}
.profile {
  display: flex;
  align-items: center;
  h2 {
    cursor: pointer;
  }
}

//
// mediaQueries
// --------------------------------

@include responsive.tablet-only() {
  .container {
    display: flex;
    justify-content: space-between;
  }

  .hamburger {
    display: flex;
  }

  .menu {
    display: block;
    margin-left: 10px;
    > a {
      svg {
        width: 31px;
        height: 30px;
      }
    }
  }

  .nav {
    display: none;
  }

  .searchContainer {
    display: flex;
    align-self: center;
  }
}

@include responsive.mobile-only() {
  .container {
    display: grid;
    grid-template-columns: 0.5fr 2fr 0.5fr;
    grid-template-rows: auto;
    grid-template-areas: 'menu actions hamburger';
  }

  .hamburger {
    grid-area: hamburger;
    display: flex;
    justify-self: end;
  }

  .menu {
    display: flex;
    grid-area: menu;
    padding-left: 6px;

    > a {
      svg {
        width: 31px;
        height: 30px;
      }
    }
  }

  .nav {
    display: none;
  }

  .actions {
    grid-area: actions;
    justify-self: center;
  }
}
