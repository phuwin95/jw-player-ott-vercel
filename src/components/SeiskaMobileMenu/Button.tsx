import React, { ReactNode } from 'react';
import styld from 'styled-components';

import Link from './NextLink';

import type { ThemeInterface } from '#types/theme';

interface ButtonElProps {
  children: ReactNode | Array<ReactNode>;
  className?: string;
  disabled?: boolean;
  href?: string;
  id?: string;
  onClick?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
  /**
   * if button is of type submit, the form will be submitted when clicked.
   */
  submit?: boolean;
  /**
   * Attribute specifies where to display the response after submitting the form
   * @example '_blank'
   */
  target?: string;
  testId?: string;
  type?: 'button' | 'link';
}

export interface ButtonProps extends ButtonElProps {
  border?: string;
  borderRadius?: string;
  height?: string;
  /**
   * Adds custom colors, background, borders, etc. depending on the theme.
   */
  intent: 'primary' | 'optional' | 'default' | 'blue' | 'green' | 'grey' | 'katsoPrimary' | 'black';
  /**
   * Adds custom padding style depending on prop size
   * @example if size === large: '0.5rem 1rem',
   * @default: '0.375rem 0.75rem',
   */
  size?: 'large' | 'default';
  width?: string;
  /**
   * Defines the relationship between a linked resource and the current document
   * @example 'noopener noreferrer
   */
  rel?: string;
  noBorder?: boolean;
  /**
   * if true, margin: 0, if false, margin: 1em 1em 1em 0
   * @default false
   */
  noMargin?: boolean;
  /**
   * Add display block style to button
   * if true, display: block, if false, display: inline-block
   * @default false
   */
  displayBlock?: boolean;
  /**
   * Add custom padding to the button
   * @default '10px 20px'
   */
  padding?: string;
  /**
   * If true, the button is loading, cursor is wait
   * @default false
   */
  isLoading?: boolean;
}

const getBackground = (
  props: ButtonProps & {
    theme: ThemeInterface;
  },
) => {
  const colors = {
    primary: props.theme?.palette?.primary.main,
    default: props.theme?.palette?.secondary.main,
    blue: props?.theme?.palette?.optional.main,
    green: props?.theme?.palette?.tertiary.main,
    katsoPrimary: 'white',
    grey: '#cccccc',
    optional: props?.theme?.palette?.optional.main,
    black: props?.theme?.palette?.primary.main,
  };
  return colors[props.intent ?? 'default'] ?? 'transparent';
};

const getBorder = (
  props: ButtonProps & {
    theme: ThemeInterface;
  },
) => {
  const borderColor: Record<string, string> = {
    grey: '#999999',
    default: props.theme?.palette?.secondary.main || '',
    primary: props.theme?.palette?.primary.main || '',
    blue: props?.theme?.palette?.optional.main || '',
    green: props?.theme?.palette?.tertiary.main || '',
    katsoPrimary: 'white',
    black: props?.theme?.palette?.primary.main || '',
  };
  return borderColor[props.intent ?? 'default'] || props.border || 'none';
};

const getColor = (
  props: ButtonProps & {
    theme: ThemeInterface;
  },
) => {
  const colors: Record<string, string> = {
    grey: '#666666',
    default: props.theme.textBlack || '',
    katsoPrimary: props.theme.colors.katsoViolet || '',
    primary: props.theme?.palette?.primary.bgColor || '',
  };
  return colors[props.intent ?? 'default'] || props.theme?.palette?.primary.bgColor || '';
};

const getHoverBackground = (
  props: ButtonProps & {
    theme: ThemeInterface;
  },
) => {
  const colors: Record<string, string> = {
    primary: props.theme.palette?.primary.dark || '',
    default: props.theme.palette?.secondary.dark || '',
    blue: props.theme.palette?.optional.dark || '',
    green: props.theme.palette?.tertiary.dark || '',
    katsoPrimary: props.theme.colors?.placeholderGray || '',
    grey: '',
    optional: props.theme.palette?.optional.dark || '',
    black: props.theme.palette?.primary.dark || '',
  };
  return colors[props.intent ?? 'default'] || 'transparent';
};

const getPadding = (props: ButtonProps) => {
  const sizes: Record<string, string> = {
    large: '0.5rem 1rem',
    default: '0.375rem 0.75rem',
  };
  return sizes[props.size ?? 'default'];
};

const ButtonEl = (props: ButtonElProps) => {
  const {
    children,
    className = undefined,
    onClick = undefined,
    type = 'button',
    href = undefined,
    disabled = false,
    submit = false,
    id = '',
    testId = '',
    target = '',
  } = props;
  return (
    <>
      {type === 'link' ? (
        <Link className={className} onClick={onClick} href={href} disabled={disabled} id={id} testId={testId} target={target}>
          {children}
        </Link>
      ) : (
        <button
          className={className}
          // @ts-ignore
          onClick={onClick}
          type={submit ? 'submit' : 'button'}
          disabled={disabled}
          id={id}
          data-testid={testId}
        >
          {children}
        </button>
      )}
    </>
  );
};

const Button = styld(ButtonEl)<ButtonProps>`
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
  user-select: none;
  font-size: 1rem;
  text-transform: uppercase;
  border-radius: ${({ borderRadius }) => borderRadius};
  border: ${(props) => getBorder(props)};
  line-height: 1.12;
  font-family: ${({ theme }) => theme.fontHeading};
  font-weight: 800;
  height: ${({ height }) => height};
  width: ${({ width }) => width};
  background: ${(props) => getBackground(props)};
  color: ${(props) => getColor(props)};
  padding: ${(props) => getPadding(props)};
  transition: all 0.2s ease-in-out;

  &:hover {
    background: ${(props) => getHoverBackground(props)};
  }
  &:disabled,
  &[disabled] {
    border: 1px solid #999999;
    background-color: #cccccc;
    color: #666666;
    cursor: not-allowed;
  }
`;

export default Button;
