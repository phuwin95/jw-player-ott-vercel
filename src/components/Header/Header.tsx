import React, { ReactNode, useState } from 'react';
import classNames from 'classnames';
import { useTranslation } from 'react-i18next';

import styles from './Header.module.scss';

import AccountCircle from '#src/icons/AccountCircle';
import SearchBar, { Props as SearchBarProps } from '#components/SearchBar/SearchBar';
import Logo from '#components/Logo/Logo';
import Link from '#components/Link/Link';
import Menu from '#src/icons/Menu';
import Hamburger from '#src/icons/Hamburger';
import Button from '#components/Button/Button';
import Popover from '#components/Popover/Popover';
import UserMenu from '#components/UserMenu/UserMenu';
import useBreakpoint, { Breakpoint } from '#src/hooks/useBreakpoint';
import IconButton from '#components/IconButton/IconButton';
import Language from '#src/icons/Language';
import LanguageMenu from '#components/LanguageMenu/LanguageMenu';
import type { LanguageDefinition } from '#src/i18n/config';
import Panel from '#components/Panel/Panel';
import type { Profile } from '#types/account';
import ProfileCircle from '#src/icons/ProfileCircle';
import CloseIcon from '#src/icons/Close';
import type { AccessModel } from '#types/Config';
import SeiskaChevronLeft from '#src/icons/SeiskaChevronLeft';

type TypeHeader = 'static' | 'fixed';

type Props = {
  headerType?: TypeHeader;
  onMenuButtonClick?: () => void;
  onPreviousButtonClick?: () => void;
  logoSrc?: string | null;
  searchBarProps: SearchBarProps;
  searchEnabled: boolean;
  hamburgerMenuOpen: boolean;
  onLoginButtonClick?: () => void;
  onSignUpButtonClick?: () => void;
  openUserMenu: () => void;
  closeUserMenu: () => void;
  openLanguageMenu: () => void;
  closeLanguageMenu: () => void;
  openHamburgerMenu: () => void;
  closeHamburgerMenu: () => void;
  children?: ReactNode;
  isLoggedIn: boolean;
  userMenuOpen: boolean;
  languageMenuOpen: boolean;
  canLogin: boolean;
  showPaymentsMenuItem: boolean;
  supportedLanguages: LanguageDefinition[];
  currentLanguage: LanguageDefinition | undefined;
  onLanguageClick: (code: string) => void;
  currentProfile?: Profile;
  profiles?: Profile[];
  profilesEnabled?: boolean;
  accessModel?: AccessModel;
  isHome?: boolean;
};

const Header: React.FC<Props> = ({
  children,
  logoSrc,
  searchBarProps,
  searchEnabled,
  onLoginButtonClick,
  onSignUpButtonClick,
  isLoggedIn,
  userMenuOpen,
  languageMenuOpen,
  openUserMenu,
  closeUserMenu,
  openLanguageMenu,
  closeLanguageMenu,
  openHamburgerMenu,
  closeHamburgerMenu,
  hamburgerMenuOpen,
  canLogin = false,
  showPaymentsMenuItem,
  supportedLanguages,
  currentLanguage,
  onLanguageClick,
  currentProfile,
  profiles,
  profilesEnabled,
  accessModel,
  onMenuButtonClick,
  onPreviousButtonClick,
  isHome,
}) => {
  const { t } = useTranslation('menu');
  const [logoLoaded, setLogoLoaded] = useState(false);
  const toggleMenu = () => {
    document.body.style.overflow = hamburgerMenuOpen ? 'auto' : 'hidden';
    if (hamburgerMenuOpen) closeHamburgerMenu();
    else openHamburgerMenu();
  };
  const breakpoint = useBreakpoint();

  // only show the language dropdown when there are other languages to choose from
  const showLanguageSwitcher = supportedLanguages.length > 1;

  const renderSearch = () => {
    if (!searchEnabled) return null;

    return (
      <div className={styles.searchContainer}>
        <SearchBar {...searchBarProps} />
      </div>
    );
  };

  const renderUserActions = () => {
    if (!canLogin || breakpoint <= Breakpoint.sm) return null;

    return isLoggedIn ? (
      <React.Fragment>
        <IconButton className={classNames(styles.iconButton, styles.actionButton)} aria-label={t('open_user_menu')} onClick={openUserMenu}>
          {currentProfile?.avatar_url ? <ProfileCircle src={currentProfile.avatar_url} alt={currentProfile.name} /> : <AccountCircle />}
        </IconButton>
        <Popover isOpen={userMenuOpen} onClose={closeUserMenu}>
          <Panel>
            <UserMenu
              onClick={closeUserMenu}
              showPaymentsItem={showPaymentsMenuItem}
              small
              accessModel={accessModel}
              currentProfile={currentProfile}
              profilesEnabled={profilesEnabled}
              profiles={profiles}
            />
          </Panel>
        </Popover>
      </React.Fragment>
    ) : (
      <div className={styles.buttonContainer}>
        <Button onClick={onLoginButtonClick} label={t('sign_in')} />
        <Button variant="contained" color="primary" onClick={onSignUpButtonClick} label={t('sign_up')} />
      </div>
    );
  };

  const renderLanguageDropdown = () => {
    if (!showLanguageSwitcher) return null;

    return (
      <React.Fragment>
        <IconButton className={classNames(styles.iconButton, styles.actionButton)} aria-label={t('select_language')} onClick={openLanguageMenu}>
          <Language />
        </IconButton>
        <Popover isOpen={languageMenuOpen} onClose={closeLanguageMenu}>
          <Panel>
            <LanguageMenu
              onClick={(code) => {
                onLanguageClick(code);
                closeLanguageMenu();
              }}
              languages={supportedLanguages}
              currentLanguage={currentLanguage}
            />
          </Panel>
        </Popover>
      </React.Fragment>
    );
  };

  return (
    <header className={styles.container}>
      <div className={styles.menu}>
        {isHome ? (
          <Link to="/p/gRb8jEQx" aria-label="all-series" onClick={onMenuButtonClick}>
            <Menu />
          </Link>
        ) : (
          <Link aria-label="back" onClick={onPreviousButtonClick}>
            <SeiskaChevronLeft />
          </Link>
        )}
      </div>
      {logoSrc && (
        <div className={styles.brand}>
          <Logo src={logoSrc} onLoad={() => setLogoLoaded(true)} />
        </div>
      )}
      <nav className={styles.nav} aria-label="menu">
        {logoLoaded || !logoSrc ? children : null}
      </nav>
      <div className={styles.actions}>
        {renderSearch()}
        {renderLanguageDropdown()}
        {renderUserActions()}
      </div>
      <div className={styles.hamburger}>
        <IconButton className={classNames(styles.iconButton, styles.mobileOnly)} aria-label={t('open_menu')} onClick={toggleMenu}>
          {hamburgerMenuOpen ? <CloseIcon /> : <Hamburger />}
        </IconButton>
      </div>
    </header>
  );
};
export default Header;
