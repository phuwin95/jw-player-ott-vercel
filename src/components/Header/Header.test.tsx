import React from 'react';
import { render } from '@testing-library/react';

import Header from './Header';

import Button from '#components/Button/Button';

vi.mock('react-router-dom', () => ({
  NavLink: () => 'a',
}));

describe('<Header />', () => {
  test('renders header', () => {
    const playlistMenuItems = [<Button key="key" label="Home" to="/" />];
    const { container } = render(
      <Header
        onMenuButtonClick={vi.fn()}
        searchBarProps={{
          query: '',
          onQueryChange: vi.fn(),
        }}
        searchEnabled
        searchActive={false}
        onSearchButtonClick={vi.fn()}
        onCloseSearchButtonClick={vi.fn()}
        onLoginButtonClick={vi.fn()}
        userMenuOpen={false}
        openUserMenu={vi.fn()}
        closeUserMenu={vi.fn()}
        openLanguageMenu={vi.fn()}
        closeLanguageMenu={vi.fn()}
        isLoggedIn={false}
        canLogin={true}
        showPaymentsMenuItem={true}
        supportedLanguages={[]}
        currentLanguage={undefined}
        languageMenuOpen={false}
        onLanguageClick={vi.fn()}
      >
        {playlistMenuItems}
      </Header>,
    );

    expect(container).toMatchSnapshot();
  });
});
