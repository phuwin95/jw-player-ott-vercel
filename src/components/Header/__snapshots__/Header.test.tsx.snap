// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > renders header 1`] = `
<div>
  <header
    class="_header_f4f7a7 _static_f4f7a7"
  >
    <div
      class="_container_f4f7a7"
    >
      <div
        class="_menu_f4f7a7"
      >
        <div
          aria-label="open_menu"
          class="_iconButton_0fef65 _iconButton_f4f7a7"
          role="button"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="_icon_1e9999"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </div>
      </div>
      <nav
        aria-label="menu"
        class="_nav_f4f7a7"
      >
        a
      </nav>
      <div
        class="_actions_f4f7a7"
      >
        <div
          aria-label="Open search"
          class="_iconButton_0fef65 _iconButton_f4f7a7 _actionButton_f4f7a7"
          role="button"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="_icon_1e9999"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </g>
          </svg>
        </div>
        <div
          class="_buttonContainer_f4f7a7"
        >
          <button
            class="_button_f8f296 _default_f8f296 _outlined_f8f296"
          >
            <span>
              sign_in
            </span>
          </button>
          <button
            class="_button_f8f296 _primary_f8f296"
          >
            <span>
              sign_up
            </span>
          </button>
        </div>
        <div
          aria-label="open_menu"
          class="_iconButton_0fef65 _iconButton_f4f7a7 _mobileOnly_f4f7a7"
          role="button"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="_icon_1e9999"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </div>
      </div>
    </div>
    <div
      class="sc-bdnyFh gbwwrQ"
    >
      <div
        class="sc-jrsKJM iYygVd"
        id="mobile-main-menu"
      >
        <ul
          class="sc-kEqXeH cEswkV"
        >
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/tuoreimmat"
            >
              Tuoreimmat
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/suosituimmat"
            >
              Luetuimmat
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/vain-seiskassa"
            >
              Vain Seiskassa
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/kotimaa"
            >
              Kotimaa
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/ulkomaat"
            >
              Ulkomaat
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/kuninkaalliset"
            >
              Kuninkaalliset
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/rikos"
            >
              Rikos
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/urheilu"
            >
              Urheilu
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/terveys"
            >
              Terveys
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/videot"
            >
              Videot
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/horoskooppi"
            >
              Horoskooppi
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <a
              class="sc-gtsqUy UBJb sc-hKFymg iJoBMj"
              href="https://www.seiska.fi/kilpailut-ja-matkat"
            >
              Kilpailut
            </a>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <div
              class="accordion"
              data-accordion-component="Accordion"
            >
              <div
                class="accordion__item"
                data-accordion-component="AccordionItem"
              >
                <div
                  aria-level="3"
                  class="accordion__heading"
                  data-accordion-component="AccordionItemHeading"
                  role="heading"
                >
                  <div
                    aria-controls="accordion__panel-:r0:"
                    aria-disabled="false"
                    aria-expanded="false"
                    bold="true"
                    class="sc-fujznN fOREje"
                    data-accordion-component="AccordionItemButton"
                    id="accordion__heading-:r0:"
                    role="button"
                    tabindex="0"
                  >
                    Katso
                    <span
                      class="sc-eCAqax sc-jSFipO sc-iCoHzw fsqkOx dGaOsA bYhkDM"
                    />
                  </div>
                </div>
                <div
                  class="sc-pNWRh jppwjX"
                  data-accordion-component="AccordionItemPanel"
                  hidden=""
                  id="accordion__panel-:r0:"
                >
                  <a
                    class="sc-gtsqUy UBJb sc-hKFymg kUOFkD"
                    href="https://www.seiska.fi/katso"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Etusivu
                  </a>
                  <a
                    class="sc-gtsqUy UBJb sc-hKFymg kUOFkD"
                    href="https://www.seiska.fi/katso/suosituimmat"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Suosituimmat
                  </a>
                  <a
                    class="sc-gtsqUy UBJb sc-hKFymg kUOFkD"
                    href="https://www.seiska.fi/katso/artikkelit"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Artikkelit
                  </a>
                  <a
                    class="sc-gtsqUy UBJb sc-hKFymg kUOFkD"
                    href="https://www.seiska.fi/katso/ohjelmisto"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Ohjelmatiedot
                  </a>
                  <a
                    class="sc-gtsqUy UBJb sc-hKFymg kUOFkD"
                    href="https://www.seiska.fi/katso/katsojaluvut"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Katsojaluvut
                  </a>
                </div>
              </div>
            </div>
          </li>
          <li
            class="sc-iqAbyq bFPRki"
          >
            <div
              class="accordion"
              data-accordion-component="Accordion"
            >
              <div
                class="accordion__item"
                data-accordion-component="AccordionItem"
              >
                <div
                  aria-level="3"
                  class="accordion__heading"
                  data-accordion-component="AccordionItemHeading"
                  role="heading"
                >
                  <div
                    aria-controls="accordion__panel-:r1:"
                    aria-disabled="false"
                    aria-expanded="false"
                    bold="true"
                    class="sc-fujznN fOREje"
                    data-accordion-component="AccordionItemButton"
                    id="accordion__heading-:r1:"
                    role="button"
                    tabindex="0"
                  >
                    Muu
                    <span
                      class="sc-eCAqax sc-jSFipO sc-iCoHzw fsqkOx dGaOsA bYhkDM"
                    />
                  </div>
                </div>
                <div
                  class="sc-pNWRh jppwjX"
                  data-accordion-component="AccordionItemPanel"
                  hidden=""
                  id="accordion__panel-:r1:"
                >
                  <a
                    class="sc-gtsqUy UBJb sc-hKFymg kUOFkD"
                    href="https://www.seiska.fi/yhteystiedot"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Yhteystiedot
                  </a>
                  <a
                    class="sc-gtsqUy UBJb sc-hKFymg kUOFkD"
                    href="https://www.seiska.fi/info/kommentoinnin-saannot/1116080"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Kommentoinnin säännöt
                  </a>
                  <a
                    class="sc-gtsqUy UBJb sc-hKFymg kUOFkD"
                    href="https://aller.fi/meista/tietosuoja/"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Tietosuojaseloste
                  </a>
                  <a
                    class="sc-gtsqUy UBJb sc-hKFymg kUOFkD"
                    href="https://www.seiska.fi/kaupallinen-yhteistyo"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Kaupallinen yhteistyö
                  </a>
                </div>
              </div>
            </div>
          </li>
        </ul>
        <div
          class="sc-crzpnZ dTcYvA"
        >
          <a
            class="sc-gtsqUy UBJb sc-dlniIP sc-dItHI hGllfd cSZbUL"
            data-testid="user_tips_mobile"
            href="https://www.seiska.fi/contact/user_tips"
            id=""
            target="_blank"
          >
            Vinkkaa ja tienaa
          </a>
          <a
            class="sc-gtsqUy UBJb sc-dlniIP sc-dItHI dIYvlJ bKHJIv"
            data-testid=""
            href="https://www.seiska.fi/button"
            id=""
            target="_blank"
          >
            Tilaa
          </a>
          <a
            class="sc-gtsqUy UBJb sc-dlniIP sc-dItHI cRMuki eJOSzR"
            data-testid=""
            href="https://www.seiska.fi/verkkolehti"
            id=""
            target="_blank"
          >
            Digilehti
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
